core/sys/types/build.rs

```diff
@@ -2,35 +2,65 @@
 //! Builds the FFI type bindings for the common SGX SDK types
 
 use bindgen::{callbacks::ParseCallbacks, Builder};
+use once_cell::sync::Lazy;
+
+// Types that don't have an SGX qualifier.
+//
+// These types are of the form <name> and later typedefed to an `sgx_<name>`.
+//
+// ```C
+//      typedef struct _foo_name {
+//          int a;
+//          float b;
+//      } sgx_foo_name;
+// ```
+//
+// To keep the noise out of the bindings, we use the underlying type and tell
+// bindgen to map directly to `sgx_<name>` version.
+static ALLOWED_UNDERLYING_TYPES: Lazy<Vec<&str>> = Lazy::new(|| {
+    vec![
+        "_status_t",
+        "_target_info_t",
+        "_attributes_t",
+        "_report_t",
+        "_key_request_t",
+    ]
+});
```

Comment:
(may be moot with my other suggestion)

Since this is a fixed immutable list, we can just use a `const` slice
```suggestion
const ALLOWED_UNDERLYING_TYPES: &[&str] = &[
    "_status_t",
    "_target_info_t",
    "_attributes_t",
    "_report_t",
    "_key_request_t",
];
```

Reply:
Thanks!, I was sure there had to be a nicer way of making a static _list_.

I think I'll leave this declared here for adding to the `allowlist_type()`

Reply:
used constant slice